src/connection.js

```diff
@@ -585,8 +594,12 @@ class Connection extends EventEmitter {
 
   requestTimeout() {
     this.requestTimer = undefined;
-    this.messageIo.sendMessage(TYPE.ATTENTION);
-    return this.transitionTo(this.STATE.SENT_ATTENTION);
+    this.cancel();
```

Comment:
There are actually two flavors of cancel timer timeout error messages:
1) Attention sent in response to request timeout.
2) Attention sent in response to user initiated cancel()